code-for-canada/project-thundercat
# [Cancel Confirmation Message] Displaying message only if there are changes

# Description

I added logic to display the cancel confirmation message only if there are changes in the concerned form (Email Form and Task Form).

If no changes has been applied to the concerned form and you try to close your response window, it'll close without any warning, since you did not change/update anything.

However, if there are changes in any of the fields of the concerned form, it'll display the cancel confirmation popup.

The behavior is exactly the same when you edit your responses (email or task), meaning that if the content is the same as it was before you opened the edit response window, it'll close without any warning and if it's not the same, obviously it'll pop the warning message.

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**When Creating a New Task, if the form looks like this, it'll close without any warning:**

_Email_

![image](https://user-images.githubusercontent.com/23021242/56596625-e1eb5a00-65be-11e9-9df3-1177981e5601.png)

_Task_
![image](https://user-images.githubusercontent.com/23021242/56596528-b7010600-65be-11e9-81fb-fe8c24dba435.png)

**If you change any of the fields, it'll pop the following:**

_Email_

![image](https://user-images.githubusercontent.com/23021242/56596797-34c51180-65bf-11e9-82c3-47b05c670985.png)

![image](https://user-images.githubusercontent.com/23021242/56596749-1b23ca00-65bf-11e9-88c6-259fee20619a.png)

_Task_

![image](https://user-images.githubusercontent.com/23021242/56596816-40b0d380-65bf-11e9-9835-c1eded93567c.png)

![image](https://user-images.githubusercontent.com/23021242/56596827-46a6b480-65bf-11e9-9e57-5e3f12a207a6.png)

**Same behavior if you do that with the edit mode**

# Testing

Manual steps to reproduce this functionality:

1.  Start the eMIB Sample Test.
2.  Complete the pre-test process.
3.  Open the _Inbox_ tab.
4.  Create new tasks and emails and make sure that the cancel confirmation message is displayed at the right moment (without and without changes).
5. Edit tasks and emails and make sure the the cancel confirmation message is displayed at the right moment (with and without changes)

# Checklist

Applicable for all code changes.
